### DISCUSSION TITLE
Siemens S7-300 communication with PLC4J via MPI cable

### DISCUSSION THREAD

**Discussion by doleron** [apache/plc4x#1692] [Q&A]
Hi everyone!

I would like to know if PLC4J supports communication with S7 PLCs (S7-300, specifically) using MPI cable (not using the ethernet module).

If so, I would appreciate if someone can point out to any source or example of using this configuration.

PS: This question was originally posted on StackOverflow: https://stackoverflow.com/questions/78715748/siemens-s7-300-communication-with-plc4j-via-mpi-cable

**Comment by chrisdutz** [comment]
Hi Luiz,
Unfortunately we currently don't have support for Siemens plcs using a serial connection.
However I think it should probably be possible to do this, as I am sure that the old serial protocol is just the predicessor of the S7comm protocol (It feels a bit like a serial protocol adjusted to network via some bubble-wrap).
The problem is that I have no equipment to test such a thing. But I see that Cesar has already responded to you on the list.
Perhaps he can help you better.

Chris

**Reply by doleron** [reply]
Hi Chris!
Yes, Cesar's solution looks like the right way to go. I just ordered a unit of that gateway to test. Once it works, I will follow up here to work as knowledge base.

Regards, Luiz
